source/code/plugin/filter_cadvisor2mdm.rb

```diff
@@ -0,0 +1,214 @@
+                        "namespace": "Insights.Containers/node", 
+                        "dimNames": [ 
+                        "host"
+                        ], 
+                        "series": [ 
+                        { 
+                            "dimValues": [ 
+                            "%{hostvalue}"
+                            ], 
+                            "min": %{metricminvalue},
+                            "max": %{metricmaxvalue}, 
+                            "sum": %{metricsumvalue}, 
+                            "count": 1 
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+            @process_incoming_stream = check_custom_metrics_availability
+            @metrics_to_collect_hash = build_metrics_hash
+            @log.debug "After check_custom_metrics_availability process_incoming_stream #{@process_incoming_stream}"
+
+            # initialize cpu and memory limit 
+            if @process_incoming_stream
+                @cpu_limit = 0.0
+                @memory_limit = 0.0 
+                
+                begin 
+                    nodeInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo('nodes').body)
+                rescue Exception => e
+                    @log.info "Error when getting nodeInventory from kube API. Exception: #{e.class} Message: #{e.message} "
+                    ApplicationInsightsUtility.sendExceptionTelemetry(e.backtrace)
+                end
+                if !nodeInventory.nil? 
+                    cpu_limit_json = KubernetesApiClient.parseNodeLimits(nodeInventory, "capacity", "cpu", "cpuCapacityNanoCores")
+                    if !cpu_limit_json.nil? 
+                        @cpu_limit = cpu_limit_json[0]['DataItems'][0]['Collections'][0]['Value']
+                        @log.info "CPU Limit #{@cpu_limit}"
+                    else
+                        @log.info "Error getting cpu_limit"
+                    end
+                    memory_limit_json = KubernetesApiClient.parseNodeLimits(nodeInventory, "capacity", "memory", "memoryCapacityBytes")
+                    if !memory_limit_json.nil?
+                        @memory_limit = memory_limit_json[0]['DataItems'][0]['Collections'][0]['Value']
+                        @log.info "Memory Limit #{@memory_limit}"
+                    else
+                        @log.info "Error getting memory_limit"
+                    end
+                end
+            end
+        end
+
+        def build_metrics_hash
+            @log.debug "Building Hash of Metrics to Collect"
+            metrics_to_collect_arr = @metrics_to_collect.split(',').map(&:strip)
+            metrics_hash = metrics_to_collect_arr.map {|x| [x.downcase,true]}.to_h
+            @log.info "Metrics Collected : #{metrics_hash}"
+            return metrics_hash
+        end
+
+        def check_custom_metrics_availability
+            aks_region = ENV['AKS_REGION']
```

Comment:
You should check explicitly for AKS_REGION & AKS_RESOURCE_ID as you depend on both.